knoten/csm.py

```diff
@@ -280,6 +308,11 @@ def generate_bodyfixed_footprint(camera, boundary, semi_major=3396190, semi_mino
     : object
       ogr polygon
     '''
+    if semi_major is None:
+        semi_major, _ = get_radii(camera)
+    if semi_minor is None:
+        _, semi_minor = get_radii(camera)
```

Comment:
It would be very odd to pass just the semi-minor and not the semi-major. We could pass a tuple. If there's one element, assume a sphere, if there's 2+ assume semi-major, semi-minor, and ignore the rest. This would make the get_radii method work seamlessly with these methods.

Reply:
:+1: I'll get this updated.